azurerm/internal/services/network/tests/vpn_site_resource_test.go

```diff
@@ -0,0 +1,247 @@
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Network.VpnSitesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_vpn_site" {
+			continue
+		}
+
+		id, err := parse.VpnSiteID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+		if err == nil {
+			return fmt.Errorf("Network.VpnSites still exists")
+		}
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Getting on Network.VpnSites: %+v", err)
+		}
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMVpnSite_basic(data acceptance.TestData) string {
+	template := testAccAzureRMVpnSite_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_vpn_site" "test" {
+  name                = "acctest-VpnSite-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  virtual_wan_id      = azurerm_virtual_wan.test.id
+  vpn_site_link {
+    name       = "link1"
+    ip_address = "10.0.0.1"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVpnSite_complete(data acceptance.TestData) string {
+	template := testAccAzureRMVpnSite_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_vpn_site" "test" {
+  name                = "acctest-VpnSite-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  virtual_wan_id      = azurerm_virtual_wan.test.id
+  address_spaces      = ["10.0.0.0/24", "10.0.1.0/24"]
+
+  device_property {
+    vendor = "Cisco"
+    model  = "foobar"
+  }
+
+  vpn_site_link {
+    name               = "link1"
+    link_provider_name = "Verizon"
+    link_speed_mbps    = 50
+    ip_address         = "10.0.0.1"
+    bgp_property {
+      asn             = 12345
+      peering_address = "10.0.0.1"
+    }
+  }
+
+  vpn_site_link {
+    name = "link2"
+    fqdn = "foo.com"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVpnSite_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMVpnSite_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_vpn_site" "import" {
+  name                = "acctest-VpnSite-%d"
+  location            = azurerm_vpn_site.test.location
+  resource_group_name = azurerm_vpn_site.test.resource_group_name
+  virtual_wan_id      = azurerm_vpn_site.test.virtual_wan_id
+  vpn_site_link {
+    name       = "link1"
+    ip_address = "10.0.0.1"
+  }
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMVpnSite_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%d"
+  location = "%s"
+}
+
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctest-vwan-%d"
```

Comment:
I have used the mixed case for the name of vpn site resource, but not for these dependent resources, since they should be tested in their own test case. Then once some service changed the naming constraint, we can identify that from the exact test case of that affected resource, which can be easier for us to identify the cause. WDYT?